### PR TITLE
Correct "edit this page on github" links on product roots and issue creation content

### LINKED ISSUE #3523: GitHub Edit Button Links to Nonexistent Resource
**Page**: [`/start`](https://docs.amplify.aws/start/?sc_icampaign=start&sc_ichannel=docs-home/q/integration/angular)

**Feedback**:

Github edit button is not working.

### LINKED ISSUE #3535: Page Links in `Feedback`-Button Created Issues Broken/Return 404s
**Page**: [`/lib/pubsub/getting-started/q/platform/[platform]`](docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/#step-1-create-iam-policies-for-aws-iot)

**Feedback**:

-- 

The above snippet is automatically created and injected into the issue body when a user clicks on the `Feedback` button on a given docs page. The `Page` link above is broken. 

Should link to: https://docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/#step-1-create-iam-policies-for-aws-iot

Actually links to: https://github.com/aws-amplify/docs/issues/docs.amplify.aws/lib/pubsub/getting-started/q/platform/js

See also #3534 (different page, similar result).

### PR DESCRIPTION
**Pull request by tjleing**
(and other pages for which the file path is /index.mdx)

TL;DR: using Next.js `router.asPath` instead of `router.pathname` leads to inconsistencies with which URL is generated.  The correct one was used for normal pages, but `ChooseFilterPage`s had the wrong one.  This led to a widespread refactor to give better names to the two router properties, corresponding to their actual functions: `router.asPath` became `url`, and `router.pathname` became `directoryPath`, because it corresponded to where the original `.mdx` file actually lived.  Also, the links in the created "Feedback" GitHub issue needed to be `https://docs.amplify.aws/...` and not just `docs.amplify.aws/...`.

- [x] fixed! ~unfortunately `/cli` is broken right now (500) and also the directory isn't showing the selected page properly, will look into that...~
- [x] checked! ~will need to double-check cli/function by rebasing off of main when #3514 is landed; right now since the page is cli/function/function -> function/function.mdx on live it works correctly, but #3514 will break the edit button on that page temporarily.~

Fixes #3523.  Fixes #3535.

### CODE REVIEW

**Comment by tjleing** [comment]
Will not merge until #3523 is also fixed by this

**Comment by tjleing** [comment]
another small point: it looks like the React information of `child._source`, including filename and line number, are not actually present on non-debug builds.  this means that the dream of using "Edit this page" to go to a fragment instead of the (typically unhelpful) base .mdx file is a little more difficult.  options include embedding some information in the `Fragments` component somehow, or tricking React into thinking this is a debug build / running two builds to extract the line # information.

**Review by mauerbac — APPROVED** [review]
